public/content/en/basics/exceptions.md

```diff
@@ -0,0 +1,139 @@
+# Exceptions
+
+This guide is only about User-`Exceptions` - System-`Errors` are usually fatal
+and should __never__ be catched.
+
+### Catching Exception
+
+A common case for exceptions is to validate potentially invalid user input.
+Once an exception is thrown, the stack will be unwound until the first matching exception
+handler is found.
+
+```
+try
+{
+    readText("dummyFile");
+}
+catch (FileException e)
+{
+    // ...
+}
+```
+
+You can also have multiple `catch` blocks and a `finally` block that is executed
+regardless of whether an error occurred. Exception can be thrown with `throw`.
+
+```
+try
+{
+    throw new StringException("You shall not pass.");
+}
+catch (FileException e)
+{
+    // ...
+}
+catch (StringException e)
+{
+    // ...
+}
+finally
+{
+    // ...
+}
+```
+
+Remember that the scope guard is usually a better solution to the `try-finally`
+pattern.
+
+### Custom exceptions
+
+One can easily inherit from `Exception` and create custom exceptions:
+
+```
+class UserNotFoundException : Exception
+{
+    this(string msg, string file = __FILE__, size_t line = __LINE__) {
+        super(msg, file, line);
+    }
+}
+throw new UserNotFoundException("D-Man is on vacation");
+```
+
+### Enter a safe world with `nothrow`
+
+The D compiler can ensure that a function can't cause catastrophic side-effects.
+Such functions can be annotated with the `nothrow` keyword and
+the compiler will yield an error if a `nothrow` function which can throw
```

Comment:
"if a no throw function throws nonetheless". (it is not really discovered because it is just detected at runtime; if would maybe also emphasise that the compiler statically forbids to throw exception in no throw functions)